### PR TITLE
Add <StylizedBaseDialog/>

### PR DESCRIPTION
**Pull request by maddiedierker**
This adds a styled version of our [`<BaseDialog/>` component](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/BaseDialog.jsx), which has all the functionality we need, but does not include standardized styles that follow our [design system](https://www.figma.com/file/hVEnbfL3Ii13Nzsp0xdGfV/Code-Design-System-(1.0)?node-id=0%3A1). There are cases where we want `<BaseDialog/>` to continue acting as a simple container, so I think it makes sense for there to be a separate, styled version.

**In code.org design system:**
Note: The header font weight and button styles in our design system differ from our current styles. I've kept our current styles for consistency while we transition to our design system.
Screen Shot 2021-05-27 at 3 23 09 PM

**In storybook:**
Note: The "X" to close the dialog is missing in the examples below because of a styling/storybook bug separate from this change. (I believe it's because the icon is absolutely positioned; it causes some weirdness when rendering the dialog from storybook.)
Screen Shot 2021-05-27 at 3 11 40 PM

### CODE REVIEW

File: apps/src/templates/BaseDialog.jsx [hunk 1]
```diff
@@ -7,6 +7,9 @@ import {BASE_DIALOG_WIDTH} from '../constants';
  * A generic modal dialog that has an x-close in the upper right, and a
  * semi-transparent backdrop. Can be closed by clicking the x, clicking the
  * backdrop, or pressing esc.
+ *
+ * To use a styled version of this component that follows our style guide,
+ * use StylizedBaseDialog.jsx.
  */
 export default class BaseDialog extends React.Component {
```
Comment (jmkulwik) [inline apps/src/templates/BaseDialog.jsx:14]: Should this become `__DeprecatedBaseDialog`?

Comment (jmkulwik) [reply apps/src/templates/BaseDialog.jsx:14]: Oops. Nevermind. On that note, is there a way we can encourage use of StylizedBaseDialog?

Comment (jmkulwik) [reply apps/src/templates/BaseDialog.jsx:14]: Or discourage use of `BaseDialog` except where appropriate?

**Review by jmkulwik — APPROVED** [review]
Couple questions about deprecation, but they are out of scope for this change. LGTM!

File: apps/src/templates/BaseDialog.jsx (hunk 1 above)
Comment (maddiedierker) [reply apps/src/templates/BaseDialog.jsx:14]: yes to both latter questions -- i left [a PSA in #apps-dev](https://codedotorg.slack.com/archives/C0T0PNR0D/p1622653630005300) directing folks to use this dialog where possible, and i'm going to demo/PSA at next week's all-team demos

**Pull request merged**